lecture-notes/4-cor.md

```diff
@@ -139,32 +174,51 @@ Gary Bernhardt.
 ## Lecture 2
 
 Even simple software has edge cases. In CS 15, for example, one assignment
-involves writing a `delete` function to remove an element from a binary search
-tree and maintain the BST invariant. This function alone had several cases: the
-node is `NULL`; the node has no children; the node has one child; the node has
+involves writing a `remove` function to remove an element from a binary search
+tree and maintain the BST invariant.
+
+```c++
+Node* BST::remove(Node* root, int value) {
+  // ...
+}
+```
+
+This function alone had several cases: the node is `NULL`; the root has the
+expected value (and you're deleting the root); the root does not have the
+expected value; the node has no children; the node has one child; the node has
 two children. Even though you probably should write unit tests, that's a
-manageable number of cases to test manually.
+manageable number of cases to test manually... for now. Are you going to
```

Comment:
Replace "Even though…" with "It might be tempting to test those inputs manually: there aren't that many, and writing tests takes time. But are you going …"? Just rewording to flow better, no technical complaints with existing wording.